Inject non-empty release overrides earlier

rdesgroppes opened the pull request:
### Motivation
So that we don't need to think twice where/whether overridden versions were injected from environment variables.

rdesgroppes committed:
Inject non-empty release overrides earlier

So that we don't need to think twice where/whether overridden versions
were injected from environment variables.

clarkb7 commented:
should you also update `_get_release_json_value` ?

JSGette commented on tasks/libs/releasing/version.py:
```diff
@@ -370,14 +370,23 @@ def get_version_numeric_only(ctx, major_version='7'):
-        # windows runners don't accepts anything else than strings in the
-        # environment when running a subprocess.
-        return {str(k): str(v) for k, v in versions[RELEASE_JSON_DEPENDENCIES].items()}
+def load_overridden_dependencies(_):
```
Having a docstring is always nice, especially for newcomers

rdesgroppes committed:
Drop unused argument

This applies following suggestion:
https://github.com/DataDog/datadog-agent/pull/40221#discussion_r2306382723

rdesgroppes commented:
> should you also update [_get_release_json_value](https://github.com/DataDog/datadog-agent/blob/e291c293d597a1bae0dbeee1b20a144d7c62e71f/tasks/libs/releasing/json.py#L267) ? We use it in various e2e tests like
> 
> https://github.com/DataDog/datadog-agent/blob/e291c293d597a1bae0dbeee1b20a144d7c62e71f/.gitlab/e2e/e2e.yml#L56-L58

@clarkb7 `_get_release_json_value` happens to be a general purpose helper. I've added a commit that addresses your request by exposing and using a specific helper for your very need.

rdesgroppes committed:
Add docstring

This addresses following request:
https://github.com/DataDog/datadog-agent/pull/40221#discussion_r2306238024

alopezz commented on tasks/libs/releasing/version.py:
```diff
@@ -370,14 +370,30 @@ def get_version_numeric_only(ctx, major_version='7'):
+    release = load_release_json()
+    if not (dependencies := release.get(RELEASE_JSON_DEPENDENCIES)):
+        raise Exception(f"Could not find {RELEASE_JSON_DEPENDENCIES!r} in release.json")
+    env = {}
```
Nit: I'd suggest more clear/consistent naming here, such as `final_dependencies` or some such (or renaming `dependencies` to`static_dependencies` or something like that). Alternatively the name of the function could be updated to reflect that this returns an "env".

alopezz reviewed (approved):
Both comments are optional.

rdesgroppes committed:
Move `load_overridden_dependencies` to own module

This applies following suggestion:
https://github.com/DataDog/datadog-agent/pull/40221#discussion_r2466030991

rdesgroppes committed:
Use more clear/consistent namings

This addresses following suggestions:
https://github.com/DataDog/datadog-agent/pull/40221#discussion_r2466153763
